pkg/client/introduce/client.go

```diff
@@ -139,9 +181,36 @@ type options struct {
 // 			destination2,
 // 		},
 // 	})
-func (c *Client) sendProposal(_ options) error {
+func (c *Client) sendProposal(o options) error {
 	return c.handleOutbound(&introduce.Proposal{
 		Type: introduce.ProposalMsgType,
-		ID:   uuid.New().String(),
-	})
+		ID:   c.newUUID(),
+	}, o)
+}
+
+func (c *Client) saveOptions(thID string, o options) error {
+	data, err := json.Marshal(o)
+	if err != nil {
+		return fmt.Errorf("save options: %w", err)
```

Comment:
this is a little picky but probably put a more accurate message here, or no message at all by returning err as is. This error is due to marshalling, not saving.

store.Put will will return a `save`.  If you want to send `save options`, then catch the error of `c.store.Put()` below and wrap it with `save options: `

Reply:
Done!